File: be/src/main/java/nl/zzpmatcher/profile/business/ProfileManagement.java
package nl.zzpmatcher.profile.business;

import nl.zzpmatcher.profile.controller.ProfileRepository;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.RequestBody;

import static nl.zzpmatcher.utils.Validator.validate;

public class ProfileManagement {

    private final ProfileRepository profileRepository;

    public ProfileManagement(ProfileRepository profileRepository) {
        this.profileRepository = profileRepository;
    }

    public Profile updateProfile(UpdateProfileCommand updateProfileCommand) {
        validate(isUsernameEqualToLoggedOnUser(updateProfileCommand), InvalidUsernameException::new);

        final Profile profile = ProfileBuilder.build(updateProfileCommand);

        profileRepository.save(profile);

        return profile;
    }

    public Profile getProfile() {
        final String username = SecurityContextHolder.getContext().getAuthentication().getName();

        return profileRepository.findOne(username);
    }

    private boolean isUsernameEqualToLoggedOnUser(@RequestBody UpdateProfileCommand updateProfileCommand) {
        return updateProfileCommand.getUsername().equals(SecurityContextHolder.getContext().getAuthentication().getName());
    }

    public class InvalidUsernameException extends RuntimeException {
        public InvalidUsernameException() {
            super("Invalid username");
        }
    }
}

File: be/src/main/java/nl/zzpmatcher/profile/business/ProfileManagement.kt
package nl.zzpmatcher.profile.business

import nl.zzpmatcher.profile.controller.ProfileRepository
import nl.zzpmatcher.utils.Validator.validate
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.web.bind.annotation.RequestBody

class ProfileManagement(private val profileRepository: ProfileRepository) {

    fun updateProfile(updateProfileCommand: UpdateProfileCommand): Profile {
        validate(isUsernameEqualToLoggedOnUser(updateProfileCommand), { throw InvalidUsernameException() })

        val profile = ProfileBuilder.build(updateProfileCommand)

        profileRepository.save(profile)

        return profile
    }

    val profile: Profile
        get() {
            val username = SecurityContextHolder.getContext().authentication.name

            return profileRepository.findOne(username)
        }

    private fun isUsernameEqualToLoggedOnUser(@RequestBody updateProfileCommand: UpdateProfileCommand): Boolean {
        return updateProfileCommand.username == SecurityContextHolder.getContext().authentication.name
    }

    inner class InvalidUsernameException : RuntimeException("Invalid username")
}
